app/resources/v1/votes.js

```diff
@@ -6,25 +6,106 @@ const logger = require('../../../lib/logger.js')()
 exports.get = async function (req, res) {
   let ballot
   try {
-    ballot = await Vote.findAll({
-      where: {
-        voterId: {
-          [Sequelize.Op.is]: null
-        }
-      },
-      order: Sequelize.literal('random()'),
-      limit: 1
-    })
+    if (!req.user) {
+      ballot = await Vote.findAll({
+        where: {
+          voterId: {
+            [Sequelize.Op.is]: null
+          }
+        },
+        order: Sequelize.literal('random()'),
+        limit: 1
+      })
+    } else {
+      ballot = await Vote.findAll({
+        // where (submitted does not contain the user's auth0 ID, or is empty) AND (voterId is null)
+        where: {
+          [Sequelize.Op.and]: [
+            {
+              [Sequelize.Op.or]: [
+                {
+                  [Sequelize.Op.not]: {
+                    submitted: {
+                      [Sequelize.Op.contains]: [req.user.sub]
+                    }
+                  }
+                },
+                {
+                  submitted: {
+                    [Sequelize.Op.is]: null
+                  }
+                }
+              ]
+            },
+            {
+              voterId: {
+                [Sequelize.Op.is]: null
+              }
+            }
+          ]
+        },
+        order: Sequelize.literal('random()'),
+        limit: 1
+      })
+    }
   } catch (error) {
     logger.error(error)
     res.status(500).json({ status: 500, msg: 'Error fetching ballot.' })
     return
   }
+
+  if (ballot && ballot.length === 0) {
+    // no eligible streets remaining
+    res
+      .status(204)
+      .json({ status: 204, msg: 'all eligible streets have been voted on' })
```

Comment:
Let's use the same grammar here for the message: capital first letter, ends with a period.

Good choice on the 204 status code.